src/session/content/chat_action_bar.rs

```diff
@@ -78,6 +85,53 @@ mod imp {
                     }));
                 },
             );
+
+            klass.install_action("chat-action-bar.join-chat", None, move |widget, _, _| {
```

Comment:
You can use the new `install_action_async`, this way you don't need to call `spawn` anymore.